### DISCUSSION TITLE
v0.31.3

### DISCUSSION THREAD

**Discussion by rfratto** [grafana-cold-storage/agent#2977] [Announcements]
This is release `v0.31.3` of the Grafana Agent.

### Upgrading
Read the [upgrade guide](https://grafana.com/docs/agent/v0.31/upgrade-guide) for specific instructions on upgrading from older versions.

### Notable changes:

This patch release contains two bug fixes for Grafana Agent Flow components: 

* `loki.source.cloudflare`: fix issue where the zone_id argument was being ignored, and the api_token argument was being used for the zone instead. 
* `loki.source.cloudflare`: fix issue where api_token argument was not marked as a sensitive field.

### Installation:
Grafana Agent is currently distributed in plain binary form, Docker container images, a Windows installer, and a Kubernetes install script. Choose whichever fits your use-case best.

#### Kubernetes

Install directions [here.](https://grafana.com/docs/grafana-cloud/quickstart/agent-k8s/)

#### Docker container:

* https://hub.docker.com/r/grafana/agent

Docker containers are published as `grafana/agent:v0.31.3`. For Windows Docker containers, use `grafana/agent:v0.31.3-windows` instead. 

#### Windows installer

The Windows installer is provided as a [release asset](https://github.com/grafana/agent/releases/download/v0.31.3/grafana-agent-installer.exe.zip) for x64 machines.

#### Binary

We provide precompiled binary executables for the most common operating systems. Choose from the assets below for your matching operating system.

Note: ppc64le builds are currently considered secondary release targets and do not have the same level of support and testing as other platforms.

Example for the `linux` operating system on `amd64`:

```bash
# download the binary
curl -O -L "https://github.com/grafana/agent/releases/download/v0.31.3/grafana-agent-linux-amd64.zip"

# extract the binary
unzip "grafana-agent-linux-amd64.zip"

# make sure it is executable
chmod a+x "grafana-agent-linux-amd64"
```

#### `agentctl`

`agentctl`, a tool for helping you interact with the Agent, is available as a Docker image:

Docker containers are published as `grafana/agentctl:v0.31.3`. For Windows Docker containers, use `grafana/agentctl:v0.31.3-windows` instead. 

Or as a binary. Like before, choose the assets below that matches your operating system. For example, with `linux` on `amd64`:

```bash
# download the binary
curl -O -L "https://github.com/grafana/agent/releases/download/v0.31.3/grafana-agentctl-linux-amd64.zip"

# extract the binary
unzip "grafana-agentctl-linux-amd64.zip"

# make sure it is executable
chmod a+x "grafana-agentctl-linux-amd64"
```

#### `agent-operator`

`agent-operator`, a Kubernetes Operator for the Grafana Agent, is available only as a Docker image:

```bash
docker pull "grafana/agent-operator:v0.31.3"
```

<hr /><em>This discussion was created from the release <a href='https://github.com/grafana/agent/releases/tag/v0.31.3'>v0.31.3</a>.</em>

**Comment by anthonysr** [comment]
These docs still say to grab files such as "agent-linux..." however in previous releases you changed the filenames to be prepended with "grafan-agent-linux" please make sure this is fixed in future docs.

**Reply by rfratto** [reply]
Sorry, we fixed this in main but haven't brought it over to the release branch. I'll fix this release.
